charts/rancher-vsphere-csi/templates/node/daemonset.yaml

```diff
@@ -117,7 +117,15 @@ spec:
             - name: CSI_ENDPOINT
               value: unix:///csi/csi.sock
             - name: MAX_VOLUMES_PER_NODE
-              value: "59" # Maximum number of volumes that controller can publish to the node. If value is not set or zero Kubernetes decide how many volumes can be published by the controller to the node.
+              {{- if and (.Values.maxPvscsiTargetsPerVm.enabled) (gt (int .Values.csiNode.maxVolumesPerNode) 0 ) }}
```

Comment:
This appears to require both vsphere 8, and a feature gate on the CSI controller. Instead of removing the comment here, can you update it to reflect these restrictions?
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/c4513d794f3aa891e30a7e9d0a2306b5868a2646/pkg/csi/service/node.go#L37-L44

Reply:
Actually the requirement seems to be using vSphere 7 and up, although I was not able to test it in versions prior to vSphere 8. In addition, the `VsanVcMgmtConfig.xml` file might be in a different folder, depending on vCenter version: the linked issue mentions `/usr/lib/vmware-vpx/vsan-health/`, but I found it in `/usr/lib/vmware-vsan`. I will update the comment to reflect the feature gate restriction in vCenter, but I was not able to identify the exact version of vSphere which this feature is available from (suspecting vSphere 7 update 3).

Reply:
I don't know where this is authoritatively documented, I was just going off of the code comment and PR at https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1696

Reply:
Sure, I was trying to extract the information from the official vSphere documentation, but apparently this is not yet documented for CNS, although it states that 64 disks per PvSCSI controller are supported since vSphere 6.7 (https://configmax.broadcom.com/guest?vmwareproduct=vSphere&release=vSphere%206.7&categories=1-0).
The comment at https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/2800#issuecomment-1970185163 led me to believe it was possible with vSphere 7, since the file path @divyenpatel mentions only exists in vCenter appliance version 7.
Nevertheless, since we know for sure it works in vSphere 8, I will update the comments accordingly.